pkg/domain/infra/abi/containers.go

```diff
@@ -1539,10 +1539,11 @@ func (ic *ContainerEngine) ContainerStats(ctx context.Context, namesOrIds []stri
 		}
 	case len(namesOrIds) > 0:
 		containerFunc = func() ([]*libpod.Container, error) { return ic.Libpod.GetContainersByList(namesOrIds) }
-	default:
-		// No containers, no latest -> query all!
+	case options.All:
 		queryAll = true
 		containerFunc = ic.Libpod.GetAllContainers
+	default:
+		containerFunc = ic.Libpod.GetRunningContainers
```

Comment:
I think you also have to set `queryAll = true` here. That var is used to ignore errors when the ctr was removed between listing and checking stats which we should do for running containers as well.